### ISSUE TITLE
Size limit on the path of the '.aiida' folder for the daemon

### ISSUE THREAD

**Issue by greschd** [aiidateam/aiida_core#1317]
Circus complains if the absolute path to the "controller socket" (and, by extension, the ``.aiida`` folder) is too long:

```
2018-03-22 00:10:47 circus[26711] [ERROR] exception ipc path "/home/greschd/large_data/projects/aiida_tbextraction_workflows/.aiida/daemon/circus-default.controller.socket" is longer than 107 characters (sizeof(sockaddr_un.sun_path)). zmq.IPC_PATH_MAX_LEN constant can be used to check addr length (if it is defined). caught
  File "/home/greschd/.virtualenvs/aiida_tbextraction_workflows/local/lib/python2.7/site-packages/tornado/gen.py", line 307, in wrapper
    yielded = next(result)
  File "/home/greschd/.virtualenvs/aiida_tbextraction_workflows/local/lib/python2.7/site-packages/circus/arbiter.py", line 529, in start
    self.ctrl.start()
  File "/home/greschd/.virtualenvs/aiida_tbextraction_workflows/local/lib/python2.7/site-packages/circus/controller.py", line 110, in start
    self.initialize()
  File "/home/greschd/.virtualenvs/aiida_tbextraction_workflows/local/lib/python2.7/site-packages/circus/controller.py", line 80, in initialize
    self.ctrl_socket.bind(self.endpoint)
  File "zmq/backend/cython/socket.pyx", line 544, in zmq.backend.cython.socket.Socket.bind
FAILED
```

**Comment by greschd** [comment]
If that's an unavoidable limit maybe we should put the sockets somewhere outside of the ``.aiida`` folder.

**Comment by sphuber** [comment]
Ai that's bad. The whole point was that we wanted to write them somewhere that we know the aiida user has write access to. Choosing another default like `/tmp` may cause other problems. I have no idea what would work here

**Comment by greschd** [comment]
I'm not sure if you can just set the ``zmq.IPC_PATH_MAX_LEN`` to a higher value, or if that's an intrinsic system limitation.

**Comment by sphuber** [comment]
The default implementation that is being used is `cython` and that implementation in `pyzmq` uses `.so` files. The limit is set and coming from in there. The `zmq` library does expose the value of this limit through the variable `IPC_PATH_MAX_LEN` but changing it has no effect. There is also the `cffi` implementation, in which we might be able to change it, but not sure how to toggle to that one yet.

**Comment by greschd** [comment]
Ah, that sucks. What's the problem with using ``/tmp`` (or ``tempfile.mkdtemp``)?

**Comment by giovannipizzi** [comment]
By the way, this is not a zmq limitation, by a kernel limit
http://man7.org/linux/man-pages/man7/unix.7.html
https://unix.stackexchange.com/questions/367008/why-is-socket-path-length-limited-to-a-hundred-chars

I think putting them in /tmp (ideally using tempfile e.g. with a prefix `/tmp/aiida-`) could be a solution, but ensuring the permissions of group and others are set to `---`. 
I tested with this simple client/server python script http://www.velvetcache.org/2010/06/14/python-unix-sockets that indeed if read/write permissions are not present on the socket, other unix users cannot communicate on it on linux, that is consistent to what written here: 
https://stackoverflow.com/questions/35424970/unix-socket-permissions-linux
However, this also says that this is not necessarily true on other BSD implementations and one should put sockets inside a folder with proper permissions.
One safe method could be to create if needed a folder named `/tmp/aiida-{UNIXUSERNAME}/` with permissions `rwx------` (or if it exists, ensure the permissions are correct) and then put short-named sockets inside it.
BTW, I also checked on Mac and also there socket file permissions are honored (a socket file with permissions  `rwx------` cannot be written/read by other users) so probably just making sure the file is created with the right permissions is ok. One might want to use umask to have the right permissions for the file, or just change permissions as soon as the socket is created?
